profile/profiler.go

```diff
@@ -0,0 +1,286 @@
+package profile
+
+import (
+	"encoding/csv"
+	"errors"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Profiler a instance of toplev command
+type Profiler struct {
+	teardown     sync.Once
+	cmd          *exec.Cmd
+	tStart       time.Time
+	interval     uint64
+	execTime     float64
+	warmTime     float64
+	tearDownTime float64
+	outFile      string
+	cores        map[string]bool
+	bottlenecks  map[string]float64
+}
+
+// NewProfiler returns a new instance of profiler
+func NewProfiler(executionTime float64, printInterval uint64, vmNum, level int, nodes, outFile string) *Profiler {
+	profiler := new(Profiler)
+	profiler.execTime = executionTime
+	profiler.interval = printInterval
+	profiler.cores = make(map[string]bool)
+	profiler.bottlenecks = make(map[string]float64)
+
+	const coreNum = 20
+
+	if outFile == "" {
+		outFile = "profile"
+	}
+	profiler.outFile = outFile + ".csv"
+
+	profiler.cmd = exec.Command("toplev.py",
+		"-v",
+		"--no-desc",
+		"-x", ",",
+		"-l", strconv.Itoa(level),
+		"-I", strconv.FormatUint(printInterval, 10),
+		"-o", profiler.outFile)
+
+	if vmNum < coreNum {
+		profiler.cmd.Args = append(profiler.cmd.Args, "--idle-threshold", "50")
+	} else {
+		profiler.cmd.Args = append(profiler.cmd.Args, "--idle-threshold", "0")
+	}
+
+	if nodes != "" {
+		profiler.cmd.Args = append(profiler.cmd.Args, "--nodes", nodes)
```

Comment:
please write a comment what `--nodes` are here for

Reply:
Done